### DISCUSSION TITLE
Issue with jose dist's browser/node version understanding

### DISCUSSION THREAD

**Discussion by suraj-sella** [panva/jose#658] [Q&A] [answered]
I am using jose with electron react vite. While using jose.generateKeyPair('ES512'), I am observing that should return type of PublicKeyType and PrivateKeyType but it is returning CryptoKey. Why is that so? I also see that after building the project, in the main js file jose uses the dist/browser folder to generate rather than node folder.

I had made a separate small project with only node where it is working perfectly. Can someone help if electron-vite with react needs some extra configs to be changed for this to work?

This is my code that works in a file index.js:

    const jose = require('jose');
    const crypto = require('node:crypto');
    (async () => {
    
    // locksmith code
    const {
        publicKey,
        privateKey
    } = await jose.generateKeyPair('ES512');
    console.log(publicKey)
    console.log(privateKey)
    
    let result = await jose.exportSPKI(publicKey);
    result = result.replace('-----BEGIN PUBLIC KEY-----', '').replace('-----END PUBLIC KEY-----', '').replace('/\r?\n|\r/g', '');
    console.log(result)
    
    // pmcore code always in ts
    const decodedPublicKey = Buffer.from(result, 'base64');
    const importedKey = crypto.createPublicKey({
        format: 'der',
        key: decodedPublicKey,
        type: 'spki',
    });
    const data = {
        message: 'this is message',
        options: 'just for fun'
    };
    console.log(importedKey);

    const jwe = await new jose.CompactEncrypt(Buffer.from(JSON.stringify(data))).setProtectedHeader({
        alg: 'ECDH-ES',
        enc: 'A256GCM'
    }).encrypt(importedKey);
    const result2 = Buffer.from(jwe).toString('base64');
    console.log(result2);

    // locksmith code
    const {
        plaintext,
        protectedHeader
    } = await jose.compactDecrypt(Buffer.from(result2, 'base64'), privateKey);
    console.log(plaintext);
    console.log(protectedHeader);
    const final = new TextDecoder().decode(plaintext);
    console.log(final);
    })();`

same doesn't work as expected in my main.ts with error:
(node:7908) UnhandledPromiseRejectionWarning: TypeError: Key must be of type CryptoKey. Received an instance of PublicKeyObject

**Comment by panva** [comment, marked as answer]
> Issue with jose dist's browser/node version understanding

If you get the WebCryptoAPI based build in electron's [main process](https://www.electronjs.org/docs/latest/tutorial/process-model#the-main-process) then the issue lies with whatever you use to bundle your application.

If it's in the renderer process than that's intended.
